### PR TITLE
Duplicate calendar event

### LINKED ISSUE #113: Duplicate event
### Steps to reproduce
1. Click on a calender event
2. No "Store as new event" button
### Expected behaviour

In the event view/frame a button "Store new event" shall be shown besides/below the "Update" button which stores the edited event as a new one without changing the original one.
### Actual behaviour

Not available

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38109459-duplicate-event?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by rogelio-o**
## Description

Added a button to duplicate an existing calendar event.

Fixes #113 

### Type of change

- [X] New feature (non-breaking change which adds functionality)

### How to test / use your changes?

1. Create calendar event (one-time event or recurrent event)
2. Open event side editor
3. Open menu (same menu that the one to delete the event) and click on "Duplicate"
4. Event (or all the instances of the recurrent event) should be duplicated

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I signed off my changes (`git commit -sm "Your commit message"`)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by rogelio-o** [comment]
Hi @tcitworld, can you review the PR if you have any time? Thanks!

**Comment by rogelio-o** [comment]
Hi @tcitworld. Any chance to move forward with this?

**Comment by loelkes** [comment]
I would also love seeing this feature. Is there something we can help you with so that this PR can move forward? @tcitworld @ChristophWurst

**Comment by loelkes** [comment]
Maybe someone else from the team can comment on this? Or review? @jancborchardt @GretaD

**Comment by ChristophWurst** [comment]
We'll have a look. Sorry for the delay :v:

**Comment by GretaD** [comment]
I gave it a general test and all looks good apart from the opacity of the icon should be darker. 
dublicayecalendar

**Comment by tcitworld** [comment]
> the opacity of the icon should be darker.

Maybe `icon-file` with `icon-category-office`?

**Comment by raimund-schluessler** [comment]
> 
> 
> > the opacity of the icon should be darker.
> 
> Maybe `icon-file` with `icon-category-office`? For reference https://docs.nextcloud.com/server/latest/developer_manual/design/icons.html

Or we move to material-design icons and use an icon which actually does indicate what the action does 🙈😉 (not really something for this PR).

Screenshot 2021-07-02 at 12-21-17 Material Design Icons

**Comment by tcitworld** [comment]
For reference again https://github.com/nextcloud/server/issues/27735

File: src/views/EditSidebar.vue [hunk 1]
```diff
@@ -93,6 +93,9 @@
 			<ActionButton v-if="canDelete && canCreateRecurrenceException" icon="icon-delete" @click="deleteAndLeave(true)">
 				{{ $t('calendar', 'Delete this and all future') }}
 			</ActionButton>
+			<ActionButton v-if="!isNew" icon="icon-file" @click="duplicateEvent()">
```
Comment (ChristophWurst) [inline src/views/EditSidebar.vue:96]: I'd say *delete* should always be the last item in the list. Move this new action up one position

**Comment by rogelio-o** [comment]
> > the opacity of the icon should be darker.
> 
> Maybe `icon-file` with `icon-category-office`? For reference https://docs.nextcloud.com/server/latest/developer_manual/design/icons.html

Change to use `icon-category-office` instead.

**Review by ChristophWurst — APPROVED** [review]

**Comment by loelkes** [comment]
@rogelio-o Can you sign your commit so we can move on with this? Or is there something else that needs to be done?

**Comment by loelkes** [comment]
Can @tcitworld and @GretaD give some feedback? I feel like its a quite simple but highly needed feature in my case. Please let me know if there's anything that can be done or if there are open (and not mentioned) issues. Maybe @ChristophWurst ?

**Review by GretaD — APPROVED** [review]
sorry for my late feedback, it works as expected. Thanks a lot

**Comment by ChristophWurst** [comment]
> @rogelio-o Can you sign your commit so we can move on with this?

@rogelio-o could you please do this? I can wave the DCO on one-liners but here I'll have to ask for your sign-off :pray:

File: src/views/EditSidebar.vue [hunk 2]
```diff
@@ -84,6 +84,9 @@
 				:href="downloadURL">
 				{{ $t('calendar', 'Download') }}
 			</ActionLink>
+			<ActionButton v-if="!isNew" icon="icon-category-office" @click="duplicateEvent()">
```
Comment (raimund-schluessler) [inline src/views/EditSidebar.vue:87]: We moved to material design icons in the meantime, so this icon needs to be adjusted. I would propose
https://materialdesignicons.com/icon/content-duplicate

**Comment by zgjavi** [comment]
Any progress on this?

**Pull request closed without merging**